internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -335,49 +335,126 @@ func resourceDatabricksWorkspace() *pluginsdk.Resource {
 				},
 			},
 
+			"enhanced_security_compliance": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"automatic_cluster_update_enabled": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"compliance_security_profile_enabled": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"compliance_security_profile_standards": {
+							Type:         pluginsdk.TypeSet,
+							Optional:     true,
+							RequiredWith: []string{"enhanced_security_compliance.0.compliance_security_profile_enabled"},
+							Elem: &pluginsdk.Schema{
+								Type: pluginsdk.TypeString,
+								ValidateFunc: validation.StringInSlice([]string{
+									string(workspaces.ComplianceStandardHIPAA),
+									string(workspaces.ComplianceStandardPCIDSS),
```

Comment:
There is a logic to imply `NONE` if `compliance_security_profile_standards` config is set to empty array (see `func expandWorkspaceEnhancedSecurity` at `internal/services/databricks/databricks_workspace_resource.go` line 1247).

This is to avoid the ambiguity of `[]` and `["NONE"]` meaning the same thing.

I will add an acctest case to cover this scenario.